editors/sc-ide/core/settings/default_themes.cpp

```diff
@@ -88,53 +88,53 @@ void Theme::fillDefault()
-    int value = bkg.value();
-    if (value > 40)
-        bkg.setHsv(bkg.hue(), bkg.saturation(), value - 11);
-    else
-        bkg.setHsv(bkg.hue(), bkg.saturation(), value + 20);
-    add("currentLine", QColor(Qt::black), bkg.toRgb());
-    add("searchResult",
-               appPlt.color(QPalette::HighlightedText).darker(200),
-               appPlt.color(QPalette::Highlight).darker(200));
-    add("matchingBrackets", QColor("#2bc93d"), Qt::yellow, true);
-    add("mismatchedBrackets", Qt::white, QColor(150,0,0));
-    add("evaluatedCode", Qt::black, QColor("#F8A200"));
-
-    QPalette plt(QApplication::palette());
-    QColor base = plt.color(QPalette::Base);
-    QColor text = plt.color(QPalette::Text);
-    int shade = (base.red() + base.green() + base.blue() < 380) ? 160 : 100;
-
-    QColor whitespace_color((base.red() + text.red()) / 2,
-                            (base.green() + text.green()) / 2,
-                            (base.blue() + text.blue()) / 2);
-
-    add("whitespace", whitespace_color);
-    add("keyword", QColor(0,0,230).lighter(shade),
-                                    QColor(Qt::transparent), true);
-    add("built-in", QColor(51,51,191).lighter(shade));
-    add("env-var", QColor(140,70,20).lighter(shade));
-    add("class", QColor(0,0,210).lighter(shade));
-    add("number", QColor(152,0,153).lighter(shade));
-    add("symbol", QColor(0,115,0).lighter(shade));
-    add("string", QColor(95,95,95).lighter(shade));
-    add("char", QColor(0,115,0).lighter(shade));
-    add("comment", QColor(191,0,0).lighter(shade));
-    add("primitive", QColor(51,51,191).lighter(shade));
-    add("lineNumbers", plt.color(QPalette::ButtonText),
-                                        plt.color(QPalette::Mid));
-    add("selection", plt.color(QPalette::HighlightedText),
-                                      plt.color(QPalette::Highlight));
-    add("postwindowtext", plt.color(QPalette::ButtonText));
-    add("postwindowerror", QColor(209, 28, 36));
-    add("postwindowwarning", QColor(165, 119, 6));
-    add("postwindowsuccess", QColor(115, 138, 5));
-    add("postwindowemphasis", Qt::black, Qt::transparent, true);
+    QColor background("#ffffff");
+    QColor white("#ffffff");
+    QColor black("#000000");
+    QColor gray("#888888");
+    QColor gray_current_line("#f4f4f4");
+    QColor orange("#F8A200");
+    QColor yellow("#ffff00");
+    QColor green_matched_brackets("#2bc93d");
+    QColor red_mismatched_brackets("#960000");
+    QColor gray_line_numbers("#b8b8b8");
+    QColor pale_blue("#3333bf");
+    QColor blue_keyword("#0000e6");
+    QColor blue_class("#0000d2");
+    QColor blue_selection("#308cc6");
+    QColor brown("#8c4614");
+    QColor purple("#980099");
+    QColor green("#007300");
+    QColor gray_string("#5f5f5f");
+    QColor red_comment("#bf0000");
+    QColor red_post_window("#d11c24");
+    QColor orange_post_window("#a57706");
+    QColor yellow_post_window("#738a05");
```

Comment:
i adopted this technique when working on the default theme and i'm pretty happy with it personally. i assign each color to a variable so that colors can be easily reused across different fields, and i give them names like `red_comment` so it's quick to mentally associate them with colors that actually appear in the theme. i'm doing this for specific reasons, but if you really dislike it then i'm fine with changing it.

Reply:
you might have a point about reuse, except none of these are being reused but black and white, which can be replaced with existing global constants.

the name "red_comment" isn't adding anything, because the purpose would already be clear from the line `add("comment", QColor("#bf0000")); // red`. i don't see how that makes mental association any more difficult. and now a reader doesn't need to jump up 30 lines to find exactly what shade of red is associated with comments.